modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/extension/SqlExtension.java

```diff
@@ -0,0 +1,132 @@
+package org.apache.ignite.internal.processors.query.calcite.extension;
+
+import java.util.List;
+import java.util.Set;
+import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptTable;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.internal.processors.query.calcite.exec.ExecutionContext;
+import org.apache.ignite.internal.processors.query.calcite.exec.rel.Node;
+import org.apache.ignite.internal.processors.query.calcite.metadata.ColocationGroup;
+import org.apache.ignite.internal.processors.query.calcite.prepare.PlannerPhase;
+import org.apache.ignite.internal.processors.query.calcite.rel.IgniteRel;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Entry point to extend current sql engine with external storage or even custom execution.
+ */
+public interface SqlExtension {
+    /**
+     * Returns the name of the current extension.
+     *
+     * <p>This name will be used to distinguish between different
+     * extensions. Also the {@link CatalogUpdateListener} will register
+     * provided catalog with the same name.
```

Comment:
But there is nothing about names in CatalogUpdateListener.

Reply:
Don't sure I understand you correctly. The `CatalogUpdateListener` is a simple functional interface. It's a matter of particular implementation how provided catalog will be registered within Ignite's internal schema holder.